From 60c719220dbe7347d3a6cb2fb63be062d727f459 Mon Sep 17 00:00:00 2001
From: Anish Trivedi <anish@freescale.com>
Date: Fri, 29 Apr 2011 11:03:13 -0500
Subject: [PATCH 909/998] ENGR00142729 SRTC: Add missing delay for writes to LP registers

Writes to LP domain registers take 3 clock cycle of the CKIL clock.
There were some writes in the driver that did not have this necessary delay.

Not all writes require this delay. The delay simply needs to be added before
the next read or return to upper layers of the driver.

Signed-off-by: Anish Trivedi <anish@freescale.com>
---
 drivers/rtc/rtc-mxc_v2.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/rtc/rtc-mxc_v2.c b/drivers/rtc/rtc-mxc_v2.c
index 4d2317b..13d6144 100644
--- a/drivers/rtc/rtc-mxc_v2.c
+++ b/drivers/rtc/rtc-mxc_v2.c
@@ -272,6 +272,7 @@ static irqreturn_t mxc_rtc_interrupt(int irq, void *dev_id)
 	/* clear interrupt status */
 	__raw_writel(lp_status, ioaddr + SRTC_LPSR);
 
+	rtc_write_sync_lp(ioaddr);
 	rtc_update_irq(pdata->rtc, 1, events);
 	return IRQ_HANDLED;
 }
@@ -327,6 +328,7 @@ static int mxc_rtc_ioctl(struct device *dev, unsigned int cmd,
 			pdata->irq_enable = false;
 		}
 		__raw_writel(lp_cr, ioaddr + SRTC_LPCR);
+		rtc_write_sync_lp(ioaddr);
 		spin_unlock_irqrestore(&rtc_lock, lock_flags);
 		return 0;
 
@@ -339,6 +341,7 @@ static int mxc_rtc_ioctl(struct device *dev, unsigned int cmd,
 		lp_cr = __raw_readl(ioaddr + SRTC_LPCR);
 		lp_cr |= SRTC_LPCR_ALP | SRTC_LPCR_WAE;
 		__raw_writel(lp_cr, ioaddr + SRTC_LPCR);
+		rtc_write_sync_lp(ioaddr);
 		spin_unlock_irqrestore(&rtc_lock, lock_flags);
 		return 0;
 
-- 
1.7.4.1

